skbio/sequence/_genetic_code.py

```diff
@@ -8,588 +8,776 @@
-            self._blocks = blocks
-            return self._blocks
-
-    blocks = property(_get_blocks)
+        References
+        ----------
+        .. [1] http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi
 
-    def __str__(self):
-        """Returns code_sequence that constructs the GeneticCode
-        """
-        return self.code_sequence
+        Examples
+        --------
+        Get the NCBI thraustochytrium mitochondrial genetic code (23):
+
+        >>> tmgc = GeneticCode.from_ncbi(23)
+        >>> tmgc.name
+        'Thraustochytrium Mitochondrial'
 
-    def __repr__(self):
-        """Returns reconstructable representation of the GeneticCode
         """
-        return 'GeneticCode(%s)' % str(self)
+        if table_id not in _ncbi_genetic_codes:
+            raise ValueError(
+                "`table_id` must be one of %r, not %r"
+                % (sorted(_ncbi_genetic_codes), table_id))
+        return _ncbi_genetic_codes[table_id]
 
-    def __eq__(self, other):
-        """ Allows two GeneticCode objects to be compared to each other.
+    @classproperty
+    def reading_frames(cls):
+        """Six possible reading frames.
+
+        Reading frames are ordered:
+
+        * 1 (forward)
+        * 2 (forward)
+        * 3 (forward)
+        * -1 (reverse)
+        * -2 (reverse)
+        * -3 (reverse)
+
+        This property can be passed into
+        ``GeneticCode.translate(reading_frame)``.
+
+        Returns
+        -------
+        list (int)
+            Six possible reading frames.
 
-        Two GeneticCode objects are equal if they have equal code_sequences.
         """
-        if not isinstance(other, GeneticCode):
-            return False
-        return self.code_sequence == other.code_sequence
+        return [1, 2, 3, -1, -2, -3]
 
-    def __ne__(self, other):
-        """Required in Py2."""
-        return not self == other
+    @property
+    def name(self):
+        """Genetic code name.
 
-    def __getitem__(self, item):
-        """Returns amino acid corresponding to codon, or codons for an aa.
+        This is simply metadata and does not affect the functionality of the
+        genetic code itself.
+
+        Returns
+        -------
+        str
+            Genetic code name.
 
-        Returns [] for empty list of codons, 'X' for unknown amino acid.
         """
-        item = str(item)
-        if len(item) == 1:  # amino acid
-            return self.synonyms.get(item, [])
-        elif len(item) == 3:  # codon
-            key = item.upper()
-            key = key.replace('U', 'T')
-            return self.codons.get(key, 'X')
-        else:
-            raise InvalidCodonError("Codon or aa %s has wrong length" % item)
-
-    def translate(self, nucleotide_sequence, start=0):
-        """Translate nucleotide to protein sequence
+        return self._name
+
+    def __init__(self, amino_acids, starts, name=''):
```

Comment:
The codon order is defined in the NCBI link you couldn't get to load. Thanks for catching this, I added the explicit ordering to the *Notes* section in case the ordering at the link changes.